overlord/devicestate/devicemgr.go

```diff
@@ -989,6 +989,55 @@ func (m *DeviceManager) ensureSeeded() error {
 	return nil
 }
 
+var processAutoImportAssertionsImp = processAutoImportAssertions
+
+// ensureAutoImportAssertions makes sure that auto import assertions
+// get processed. assertion should be processed while seeding is in progress
+func (m *DeviceManager) ensureAutoImportAssertions() {
+	if release.OnClassic {
+		return
+	}
+
+	if m.earlyDeviceSeed == nil {
```

Comment:
this should be done after taking the lock otherwise is potentially racy, no?

Reply:
also populateStateFromSeedImpl resets this, so I fear this doesn't quite work in the non-preseeding case

Reply:
I have moved it after lock.
Strange I have tested this in non-preseeded cases and it does pick cache.
I will retest again both scenarios.